src/NServiceBus.Core/DelayedDelivery/TimeoutManager/StoreTimeoutBehavior.cs

```diff
@@ -65,8 +65,7 @@ public StoreTimeoutBehavior(ExpiredTimeoutsPoller poller, IDispatchMessages disp
                 {
                     var outgoingMessage = new OutgoingMessage(context.MessageId, data.Headers, data.State);
                     var transportOperation = new TransportOperation(outgoingMessage, new UnicastAddressTag(data.Destination));
-                    var transportTransaction = context.Context.GetOrCreate<TransportTransaction>();
-                    await dispatcher.Dispatch(new TransportOperations(transportOperation), transportTransaction, context.Context).ConfigureAwait(false);
+                    await dispatcher.Dispatch(new TransportOperations(transportOperation), context.TransportTransaction, context.Context).ConfigureAwait(false);
```

Comment:
Yes but there was not reason not to make the code simpler in one fell swoop. It is also in line with how the Gateway does it right now.